Deprecated warnings with the latest framework-arduinoespressif32 (idf-release/v4.2 branch)

Following compilation warning occurs when using the latest framework-arduinoespressif32 (idf-release/v4.2 branch) and PlatformIO. This is because `esp_intr.h` has been deprecated since ESP-IDF v4.0.

I'll create a PR to fix this issue.

Warnings
``` sh
In file included from .pio\libdeps\m5paper-test\FastLED\src\platforms/esp/32/clockless_rmt_esp32.h:116,
                 from .pio\libdeps\m5paper-test\FastLED\src\platforms/esp/32/fastled_esp32.h:8,
                 from .pio\libdeps\m5paper-test\FastLED\src\platforms.h:36,
                 from .pio\libdeps\m5paper-test\FastLED\src\FastLED.h:52,
                 from .pio\libdeps\m5paper-test\FastLED\src\FastLED.cpp:2:
.pio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/esp32/include/esp_intr.h:16:2: warning: #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead [-Wcpp]
 #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead
  ^~~~~~~
```

## Code example
`platformio.ini`
``` ini
[platformio]
packages_dir = .pio/packages

[env:m5paper-test]
platform = espressif32
board = m5stack-fire
framework = arduino
platform_packages =
    toolchain-xtensa32 @ https://bintray.com/platformio/tool-packages/download_file?file_path=53bcf98-toolchain-xtensa32-windows-2.80400.2020.tar.gz
    framework-arduinoespressif32 @ https://github.com/espressif/arduino-esp32.git#idf-release/v4.2
build_flags =
    -DESP32=1
    -DARDUINO_ARCH_ESP32=1
lib_deps =
  FastLED/FastLED
```

`src/main.cpp`
``` cpp
#include <FastLED.h>

void setup(void)
{
}

void loop(void)
{
}
```

## Build log of `pio run`
``` 
PS C:\Users\estshorter\src\esp> pio run
Processing m5paper-test (platform: espressif32; board: m5stack-fire; framework: arduino)
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/m5stack-fire.html
PLATFORM: Espressif 32 (2.1.0) > M5Stack FIRE
HARDWARE: ESP32 240MHz, 6.25MB RAM, 16MB Flash
DEBUG: Current (esp-prog) External (esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-arduinoespressif32 0.0.0+sha.7d3f499
 - tool-esptoolpy 1.30000.201119 (3.0.0)
 - toolchain-xtensa32 2.80400.2020 (8.4.0)
LDF: Library Dependency Finder -> http://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 30 compatible libraries
Scanning dependencies...
Dependency Graph
|-- <FastLED> 3.4.0     
|   |-- <SPI> 1.0       
Building in release mode
Compiling .pio\build\m5paper-test\libe35\FastLED\FastLED.cpp.o
Compiling .pio\build\m5paper-test\libe35\FastLED\bitswap.cpp.o
Compiling .pio\build\m5paper-test\libe35\FastLED\colorpalettes.cpp.o
Compiling .pio\build\m5paper-test\libe35\FastLED\colorutils.cpp.o
Compiling .pio\build\m5paper-test\libe35\FastLED\hsv2rgb.cpp.o
Compiling .pio\build\m5paper-test\libe35\FastLED\lib8tion.cpp.o
Compiling .pio\build\m5paper-test\libe35\FastLED\noise.cpp.o
Compiling .pio\build\m5paper-test\libe35\FastLED\platforms.cpp.o
Compiling .pio\build\m5paper-test\libe35\FastLED\platforms\esp\32\clockless_rmt_esp32.cpp.o
Compiling .pio\build\m5paper-test\libe35\FastLED\power_mgt.cpp.o
Compiling .pio\build\m5paper-test\libe35\FastLED\wiring.cpp.o
Compiling .pio\build\m5paper-test\src\main.cpp.o
Generating partitions .pio\build\m5paper-test\partitions.bin
Compiling .pio\build\m5paper-test\lib51f\SPI\SPI.cpp.o
Archiving .pio\build\m5paper-test\libFrameworkArduinoVariant.a
Compiling .pio\build\m5paper-test\FrameworkArduino\Esp.cpp.o
Compiling .pio\build\m5paper-test\FrameworkArduino\FunctionalInterrupt.cpp.o
Compiling .pio\build\m5paper-test\FrameworkArduino\HardwareSerial.cpp.o
Compiling .pio\build\m5paper-test\FrameworkArduino\IPAddress.cpp.o
Archiving .pio\build\m5paper-test\lib51f\libSPI.a
Compiling .pio\build\m5paper-test\FrameworkArduino\IPv6Address.cpp.o
In file included from .pio\libdeps\m5paper-test\FastLED\src\platforms/esp/32/clockless_rmt_esp32.h:116,
                 from .pio\libdeps\m5paper-test\FastLED\src\platforms/esp/32/fastled_esp32.h:8,
                 from .pio\libdeps\m5paper-test\FastLED\src\platforms.h:36,
                 from .pio\libdeps\m5paper-test\FastLED\src\FastLED.h:52,
                 from .pio\libdeps\m5paper-test\FastLED\src\FastLED.cpp:2:
.pio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/esp32/include/esp_intr.h:16:2: warning: #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead [-Wcpp]
 #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead
  ^~~~~~~
In file included from .pio\libdeps\m5paper-test\FastLED\src\platforms/esp/32/clockless_rmt_esp32.h:116,
                 from .pio\libdeps\m5paper-test\FastLED\src\platforms/esp/32/fastled_esp32.h:8,
                 from .pio\libdeps\m5paper-test\FastLED\src\platforms.h:36,
                 from .pio\libdeps\m5paper-test\FastLED\src\FastLED.h:52,
                 from .pio\libdeps\m5paper-test\FastLED\src\bitswap.cpp:2:
.pio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/esp32/include/esp_intr.h:16:2: warning: #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead [-Wcpp]
 #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead
  ^~~~~~~
In file included from .pio\libdeps\m5paper-test\FastLED\src\platforms/esp/32/clockless_rmt_esp32.h:116,
                 from .pio\libdeps\m5paper-test\FastLED\src\platforms/esp/32/fastled_esp32.h:8,
                 from .pio\libdeps\m5paper-test\FastLED\src\platforms.h:36,
                 from .pio\libdeps\m5paper-test\FastLED\src\FastLED.h:52,
                 from .pio\libdeps\m5paper-test\FastLED\src\colorpalettes.cpp:4:
.pio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/esp32/include/esp_intr.h:16:2: warning: #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead [-Wcpp]
 #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead
  ^~~~~~~
In file included from .pio\libdeps\m5paper-test\FastLED\src\platforms/esp/32/clockless_rmt_esp32.h:116,
                 from .pio\libdeps\m5paper-test\FastLED\src\platforms/esp/32/fastled_esp32.h:8,
                 from .pio\libdeps\m5paper-test\FastLED\src\platforms.h:36,
                 from .pio\libdeps\m5paper-test\FastLED\src\FastLED.h:52,
                 from .pio\libdeps\m5paper-test\FastLED\src\colorutils.cpp:7:
.pio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/esp32/include/esp_intr.h:16:2: warning: #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead [-Wcpp]
 #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead
  ^~~~~~~
Compiling .pio\build\m5paper-test\FrameworkArduino\MD5Builder.cpp.o
Compiling .pio\build\m5paper-test\FrameworkArduino\Print.cpp.o
Compiling .pio\build\m5paper-test\FrameworkArduino\Stream.cpp.o
In file included from .pio\libdeps\m5paper-test\FastLED\src\platforms/esp/32/clockless_rmt_esp32.h:116,
                 from .pio\libdeps\m5paper-test\FastLED\src\platforms/esp/32/fastled_esp32.h:8,
                 from .pio\libdeps\m5paper-test\FastLED\src\platforms.h:36,
                 from .pio\libdeps\m5paper-test\FastLED\src\FastLED.h:52,
                 from .pio\libdeps\m5paper-test\FastLED\src\hsv2rgb.cpp:4:
.pio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/esp32/include/esp_intr.h:16:2: warning: #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead [-Wcpp]
 #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead
  ^~~~~~~
In file included from .pio\libdeps\m5paper-test\FastLED\src\platforms/esp/32/clockless_rmt_esp32.h:116,
                 from .pio\libdeps\m5paper-test\FastLED\src\platforms/esp/32/fastled_esp32.h:8,
                 from .pio\libdeps\m5paper-test\FastLED\src\platforms.h:36,
                 from .pio\libdeps\m5paper-test\FastLED\src\FastLED.h:52,
                 from .pio\libdeps\m5paper-test\FastLED\src\lib8tion.cpp:3:
.pio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/esp32/include/esp_intr.h:16:2: warning: #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead [-Wcpp]
 #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead
  ^~~~~~~
In file included from .pio\libdeps\m5paper-test\FastLED\src\platforms/esp/32/clockless_rmt_esp32.h:116,
                 from .pio\libdeps\m5paper-test\FastLED\src\platforms/esp/32/fastled_esp32.h:8,
                 from .pio\libdeps\m5paper-test\FastLED\src\platforms.h:36,
                 from .pio\libdeps\m5paper-test\FastLED\src\FastLED.h:52,
                 from .pio\libdeps\m5paper-test\FastLED\src\noise.cpp:2:
.pio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/esp32/include/esp_intr.h:16:2: warning: #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead [-Wcpp]
 #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead
Compiling .pio\build\m5paper-test\FrameworkArduino\StreamString.cpp.o
  ^~~~~~~
In file included from .pio/libdeps/m5paper-test/FastLED/src/platforms/esp/32/clockless_rmt_esp32.h:116,
                 from .pio/libdeps/m5paper-test/FastLED/src/platforms/esp/32/fastled_esp32.h:8,
                 from .pio/libdeps/m5paper-test/FastLED/src/platforms.h:36,
                 from .pio/libdeps/m5paper-test/FastLED/src/FastLED.h:52,
                 from .pio\libdeps\m5paper-test\FastLED\src\platforms\esp\32\clockless_rmt_esp32.cpp:5:
.pio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/esp32/include/esp_intr.h:16:2: warning: #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead [-Wcpp]
 #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead
  ^~~~~~~
In file included from .pio\libdeps\m5paper-test\FastLED\src\platforms/esp/32/clockless_rmt_esp32.h:116,
                 from .pio\libdeps\m5paper-test\FastLED\src\platforms/esp/32/fastled_esp32.h:8,
                 from .pio\libdeps\m5paper-test\FastLED\src\platforms.h:36,
                 from .pio\libdeps\m5paper-test\FastLED\src\FastLED.h:52,
                 from .pio\libdeps\m5paper-test\FastLED\src\power_mgt.cpp:2:
.pio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/esp32/include/esp_intr.h:16:2: warning: #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead [-Wcpp]
 #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead
  ^~~~~~~
In file included from .pio\libdeps\m5paper-test\FastLED\src\platforms/esp/32/clockless_rmt_esp32.h:116,
                 from .pio\libdeps\m5paper-test\FastLED\src\platforms/esp/32/fastled_esp32.h:8,
                 from .pio\libdeps\m5paper-test\FastLED\src\platforms.h:36,
                 from .pio\libdeps\m5paper-test\FastLED\src\FastLED.h:52,
                 from .pio\libdeps\m5paper-test\FastLED\src\wiring.cpp:2:
.pio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/esp32/include/esp_intr.h:16:2: warning: #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead [-Wcpp]
 #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead
  ^~~~~~~
Compiling .pio\build\m5paper-test\FrameworkArduino\USB.cpp.o
In file included from .pio/libdeps/m5paper-test/FastLED/src/platforms/esp/32/clockless_rmt_esp32.h:116,
                 from .pio/libdeps/m5paper-test/FastLED/src/platforms/esp/32/fastled_esp32.h:8,
                 from .pio/libdeps/m5paper-test/FastLED/src/platforms.h:36,
                 from .pio/libdeps/m5paper-test/FastLED/src/FastLED.h:52,
Compiling .pio\build\m5paper-test\FrameworkArduino\USBCDC.cpp.o
                 from src\main.cpp:1:
.pio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/esp32/include/esp_intr.h:16:2: warning: #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead [-Wcpp]
 #warning esp_intr.h is deprecated, please include esp_intr_alloc.h instead
  ^~~~~~~
In file included from src\main.cpp:1:
Compiling .pio\build\m5paper-test\FrameworkArduino\WMath.cpp.o
.pio/libdeps/m5paper-test/FastLED/src/FastLED.h:14:21: note: #pragma message: FastLED version 3.004.000
 #    pragma message "FastLED version 3.004.000"
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
Compiling .pio\build\m5paper-test\FrameworkArduino\WString.cpp.o
Compiling .pio\build\m5paper-test\FrameworkArduino\base64.cpp.o
In file included from .pio/libdeps/m5paper-test/FastLED/src/FastLED.h:65,
                 from src\main.cpp:1:
.pio/libdeps/m5paper-test/FastLED/src/fastspi.h:135:23: note: #pragma message: No hardware SPI pins defined.  All SPI access will default to bitbanged output
 #      pragma message "No hardware SPI pins defined.  All SPI access will default to bitbanged output"
                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Compiling .pio\build\m5paper-test\FrameworkArduino\cbuf.cpp.o
Compiling .pio\build\m5paper-test\FrameworkArduino\esp32-hal-adc.c.o
Compiling .pio\build\m5paper-test\FrameworkArduino\esp32-hal-bt.c.o
Compiling .pio\build\m5paper-test\FrameworkArduino\esp32-hal-cpu.c.o
Compiling .pio\build\m5paper-test\FrameworkArduino\esp32-hal-dac.c.o
Compiling .pio\build\m5paper-test\FrameworkArduino\esp32-hal-gpio.c.o
Compiling .pio\build\m5paper-test\FrameworkArduino\esp32-hal-i2c.c.o
Compiling .pio\build\m5paper-test\FrameworkArduino\esp32-hal-ledc.c.o
Compiling .pio\build\m5paper-test\FrameworkArduino\esp32-hal-matrix.c.o
Compiling .pio\build\m5paper-test\FrameworkArduino\esp32-hal-misc.c.o
Compiling .pio\build\m5paper-test\FrameworkArduino\esp32-hal-psram.c.o
Compiling .pio\build\m5paper-test\FrameworkArduino\esp32-hal-rmt.c.o
Compiling .pio\build\m5paper-test\FrameworkArduino\esp32-hal-sigmadelta.c.o
Compiling .pio\build\m5paper-test\FrameworkArduino\esp32-hal-spi.c.o
Compiling .pio\build\m5paper-test\FrameworkArduino\esp32-hal-time.c.o
Compiling .pio\build\m5paper-test\FrameworkArduino\esp32-hal-timer.c.o
Compiling .pio\build\m5paper-test\FrameworkArduino\esp32-hal-tinyusb.c.o
Compiling .pio\build\m5paper-test\FrameworkArduino\esp32-hal-touch.c.o
Compiling .pio\build\m5paper-test\FrameworkArduino\esp32-hal-uart.c.o
Compiling .pio\build\m5paper-test\FrameworkArduino\libb64\cdecode.c.o
Compiling .pio\build\m5paper-test\FrameworkArduino\libb64\cencode.c.o
Compiling .pio\build\m5paper-test\FrameworkArduino\main.cpp.o
Compiling .pio\build\m5paper-test\FrameworkArduino\stdlib_noniso.c.o
Compiling .pio\build\m5paper-test\FrameworkArduino\wiring_pulse.c.o
Compiling .pio\build\m5paper-test\FrameworkArduino\wiring_shift.c.o
Archiving .pio\build\m5paper-test\libFrameworkArduino.a
Linking .pio\build\m5paper-test\firmware.elf
Building .pio\build\m5paper-test\firmware.bin
Retrieving maximum program size .pio\build\m5paper-test\firmware.elf
Checking size .pio\build\m5paper-test\firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM:   [          ]   0.2% (used 14216 bytes from 6553600 bytes)
Flash: [          ]   3.2% (used 208668 bytes from 6553600 bytes)
esptool.py v3.0
==================================================================================================== [SUCCESS] Took 12.67 seconds ====================================================================================================

```

## Environment
- PlatformIO IDE v2.2.1
- VS Code 1.52.1
- Win 10 20H2

--- a/src/platforms/esp/32/clockless_i2s_esp32.h
+++ b/src/platforms/esp/32/clockless_i2s_esp32.h
@@ -103,7 +103,13 @@ extern "C" {
 #include "driver/gpio.h"
 #include "driver/periph_ctrl.h"
 #include "rom/lldesc.h"
+#include "esp_system.h" // Load ESP_IDF_VERSION_MAJOR if exists
+// ESP_IDF_VERSION_MAJOR is defined in ESP-IDF v3.3 or later
+#if defined(ESP_IDF_VERSION_MAJOR) && ESP_IDF_VERSION_MAJOR > 3
+#include "esp_intr_alloc.h"
+#else
 #include "esp_intr.h"
+#endif
 #include "esp_log.h"
     
 #ifdef __cplusplus

--- a/src/platforms/esp/32/clockless_rmt_esp32.h
+++ b/src/platforms/esp/32/clockless_rmt_esp32.h
@@ -113,7 +113,12 @@ extern "C" {
 #endif
 
 #include "esp32-hal.h"
+// ESP_IDF_VERSION_MAJOR is defined in ESP-IDF v3.3 or later
+#if defined(ESP_IDF_VERSION_MAJOR) && ESP_IDF_VERSION_MAJOR > 3
+#include "esp_intr_alloc.h"
+#else
 #include "esp_intr.h"
+#endif
 #include "driver/gpio.h"
 #include "driver/rmt.h"
 #include "driver/periph_ctrl.h"

